pennylane_lightning/lightning_kokkos/lightning_kokkos.toml

```diff
@@ -1,124 +1,110 @@
+PhaseShift             = { properties = [ "invertible",                 "differentiable" ] }
+QubitUnitary           = { properties = [ "invertible",                 "differentiable" ] }
+Rot                    = { properties = [ "invertible",                 "differentiable" ] }
+RX                     = { properties = [ "invertible",                 "differentiable" ] }
+RY                     = { properties = [ "invertible",                 "differentiable" ] }
+RZ                     = { properties = [ "invertible",                 "differentiable" ] }
+SingleExcitationMinus  = { properties = [ "invertible",                 "differentiable" ] }
+SingleExcitationPlus   = { properties = [ "invertible",                 "differentiable" ] }
+SingleExcitation       = { properties = [ "invertible",                 "differentiable" ] }
+S                      = { properties = [ "invertible",                 "differentiable" ] }
+SWAP                   = { properties = [ "invertible",                 "differentiable" ] }
+Toffoli                = { properties = [ "invertible",                 "differentiable" ] }
+T                      = { properties = [ "invertible",                 "differentiable" ] }
 
 # Operators that should be decomposed according to the algorithm used
 # by PennyLane's device API.
 # Optional, since gates not listed in this list will typically be decomposed by
 # default, but can be useful to express a deviation from this device's regular
 # strategy in PennyLane.
-decomp = [
-        "BasisState",
-        "QubitStateVector",
-        "StatePrep",
-        "QFT",
-        "MultiControlledX",
-]
+[operators.gates.decomp]
+
+BasisState             = {}
+MultiControlledX       = {}
+QFT                    = {}
+QubitStateVector       = {}
+StatePrep              = {}
 
 # Gates which should be translated to QubitUnitary
-matrix = [
-        "ControlledQubitUnitary",
-        "ECR",
-        "SX",
-        "ISWAP",
-        "PSWAP",
-        "SISWAP",
-        "SQISW",
-        "CPhase",
-        "OrbitalRotation",
-        "QubitCarry",
-        "QubitSum",
-        "DiagonalQubitUnitary",
-        "BlockEncode",
-]
+[operators.gates.matrix]
+
+BlockEncode            = {}
+CPhase                 = {}
+DiagonalQubitUnitary   = {}
+ECR                    = {}
+ISWAP                  = {}
+OrbitalRotation        = {}
+PSWAP                  = {}
+QubitCarry             = {}
+QubitSum               = {}
+SISWAP                 = {}
+SQISW                  = {}
+SX                     = {}
+
+# Observables supported by the device
+[operators.observables]
+
+Exp                    = {}
+Hadamard               = {}
+Hamiltonian            = {}
+Hermitian              = {}
+Identity               = {}
+PauliX                 = {}
+PauliY                 = {}
+PauliZ                 = {}
+Prod                   = {}
+Projector              = {}
```

Comment:
I don't think this is supported.

Reply:
Hmm, but I thing that the Kokkos device does report this op in it `device.operations` set. I need to double-check.

Reply:
@vincentmr could you please check the purpose of [this](https://github.com/PennyLaneAI/pennylane-lightning/blob/1dae57c2ebe38793be929aa4e5c73d3ab1243b0f/pennylane_lightning/lightning_kokkos/lightning_kokkos.py#L157) ?

Reply:
I'll repost something from Slack here, for confirmation from the comp-team and to converge on a simple message for everyone who asks. 

> The TOML files are for Catalyst use, they mean there is a special gate implementation at the C++ level. The `operations` set means there is support at the Python layer (at least) and this is used by the QNode for controlling decomposition. For example `C(GlobalPhase)` is not in the L-Kokkos TOML, because there is no C++ implementation. It is in `allowed_operations` however because the device uses a mix of Python/C++ functions to apply it, and hence it is supported at the Python level.

This is outdated by this PR, but the idea remains the same: the TOML files described the C++ backend period; the `operations` and `observables` frozensets describe the capabilities of the Python device, which may surpass the C++ backend (e.g. when something is implemented in Python), or which may be surpassed by the C++ backend (e.g. when bindings do not exist and there is no corresponding Python-level functionality).

Reply:
Ah, I see. Indeed, there are two PL-L APIs: the Python and the C++ one, and they need not to be the same of cause.
The first conclusion from this: we are using our validation procedure incorrectly when it comes to observables, becuase we compare `device.observables` (Python) with `toml.observables` (C++) and treat mismatch as an error.
The second conclusion: the toml schema 2, as of now, does not describe the Python API at all, so we can not make the strict validation against Python's `device`. Ideally we would need `operators.observables.unsupported`.

I will raise this question in Slack, for now I am going to remove the unsupported observables and relax the validation procedure.

Reply:
Done:
* Ali removed non-native observables
* Sergei opened a Catalyst PR relaxing the observables validation